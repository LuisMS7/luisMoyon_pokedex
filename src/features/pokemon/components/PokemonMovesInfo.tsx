import React, { useMemo } from "react";
import { Pokemon } from "@pokemon/types/pokemon.type";
import { TagType } from "@src/types/tag.type";
import TagRow from "@src/components/TagRow";
import { NORMAL_WEIGHT } from "@lib/constants";

type PokemonMovesInfoProps = {
	pokemon: Pokemon;
	typeColor: string;
};

const PokemonMovesInfo = ({ pokemon, typeColor }: PokemonMovesInfoProps) => {
	const movesTags: TagType[] = useMemo(() => {
		return (
			pokemon.moves?.map((move) => ({
				text: move.move.name,
				backgroundColor: "white",
				borderColor: typeColor,
				textColor: "black",
				fontWeight: NORMAL_WEIGHT,
			})) || []
		);
	}, [pokemon]);

	return <TagRow tags={movesTags} />;
};

export default PokemonMovesInfo;
