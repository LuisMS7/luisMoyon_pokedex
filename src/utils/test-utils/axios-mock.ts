import client from "@lib/axios-instance";
import MockAdapter from "axios-mock-adapter";

export const mockClient = new MockAdapter(client);

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const mockGetOnSuccess = (endpoint: string, mockResponse: any) => {
	mockClient.onGet(endpoint).reply(200, mockResponse);
};

export const mockGetOnFailure = (endpoint: string, statusCode: number) => {
	mockClient.onGet(endpoint).reply(statusCode);
};
