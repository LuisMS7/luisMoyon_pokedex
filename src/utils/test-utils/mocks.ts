import { Pokemon } from "@pokemon/types/pokemon.type";

export const mockPokemon: Pokemon = {
	id: 1,
	name: "mockPokemonName",
	// eslint-disable-next-line max-len
	img: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
	types: [{ type: { name: "normal", url: "" }, slot: 1 }],
	weight: 10,
	sprites: {
		back_default:
			// eslint-disable-next-line max-len
			"https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/official-artwork/1.png",
		back_female: "",
		back_shiny: "",
		back_shiny_female: "",
		front_default: "",
		front_female: "",
		front_shiny: "",
		front_shiny_female: "",
	},
	moves: [{ move: { name: "normal", url: "" }, version_group_details: [] }],
};
