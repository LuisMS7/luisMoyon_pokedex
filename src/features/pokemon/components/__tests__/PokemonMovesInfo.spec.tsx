import React from "react";
import { render, screen } from "@testing-library/react-native";
import PokemonMovesInfo from "@pokemon/components/PokemonMovesInfo";
import { mockPokemon } from "@utils/test-utils/mocks";

describe("PokemonMovesInfo", () => {
	it("should render correctly", () => {
		render(<PokemonMovesInfo pokemon={mockPokemon} typeColor={"red"} />);
		expect(screen.toJSON()).toMatchSnapshot();
	});
});
