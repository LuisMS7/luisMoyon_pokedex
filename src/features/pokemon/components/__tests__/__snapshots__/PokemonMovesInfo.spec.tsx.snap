// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonMovesInfo should render correctly 1`] = `
<View
  style={
    {
      "alignSelf": "center",
      "flexDirection": "row",
      "flexWrap": "wrap",
      "gap": 10,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderColor": "red",
        "borderRadius": 1000,
        "borderWidth": 1,
        "justifyContent": "center",
        "paddingHorizontal": 10,
        "paddingVertical": 5,
      }
    }
  >
    <Text
      style={
        {
          "color": "black",
          "fontSize": 17.89194962785868,
          "fontWeight": "400",
        }
      }
    >
      normal
    </Text>
  </View>
</View>
`;
